payment_method_gateway.go

```diff
@@ -24,6 +24,27 @@ type PaymentMethodRequestOptions struct {
 	VerificationMerchantAccountId string `xml:"verification-merchant-account-id,omitempty"`
 }
 
+type PaymentMethodGrantRequest struct {
+	XMLName                  xml.Name `xml:"payment-method"`
+	SharedPaymentMethodToken string   `xml:"shared-payment-method-token,omitempty"`
+	AllowVaulting            bool     `xml:"allow-vaulting,omitempty"`
+	IncludeBillingPostalCode bool     `xml:"including-billing-postal-code,omitempty"`
+	RevokeAfter              string   `xml:"revoke-after,omitempty"`
+}
+
+func (g *PaymentMethodGateway) Grant(ctx context.Context, token string, paymentMethodGrantRequest *PaymentMethodGrantRequest) (*PaymentMethodNonce, error) {
+	paymentMethodGrantRequest.SharedPaymentMethodToken = token
```

Comment:
Why are we accepting a token and setting it here instead of accepting just the paymentMethodGrantRequest? If the client is creating the paymentMethodGrantRequest prior to calling this function then the client will probably set the token as well.

Reply:
Good point.